### PR TITLE
feat: task title from summary

### PR DESCRIPTION
**Pull request by matthiasn**
# Add Automatic Task Title Extraction from AI-Generated Summaries

## Overview

This PR implements automatic task title extraction from AI-generated summaries, improving the workflow for creating tasks from audio recordings or other sources where titles may be initially empty or generic.

  Changes

  1. Prompt Template Updates

  - Updated the task summary prompt in /Users/mn/github/lotti/lib/features/ai/util/preconfigured_prompts.dart to
  instruct the AI to start with a single H1 header (# Title) suggesting a concise, descriptive title for the task
  - The title should be under 80-100 characters and capture the essence of the task

  2. Automatic Title Extraction

  - Added title extraction logic in _handlePostProcessing method in
  /Users/mn/github/lotti/lib/features/ai/repository/unified_ai_inference_repository.dart
  - Uses regex pattern ^#\s+(.+) to extract the H1 title from AI responses
  - Automatically updates the task title if the current title is less than 5 characters long
  - This enables automatic title generation for tasks created from audio recordings

  3. Display Filtering

  - Implemented _filterTaskSummaryResponse method in
  /Users/mn/github/lotti/lib/features/ai/ui/ai_response_summary.dart
  - Filters out the H1 title when displaying task summaries to avoid redundancy (since the title is shown separately
  in the UI)
  - The full response including the title is preserved in the database

  4. Test Updates

  - Added comprehensive tests for title extraction functionality in unified AI inference repository tests
  - Added widget tests for H1 title filtering in the display component
  - Fixed failing tests in /Users/mn/github/lotti/test/features/ai/repository/ai_input_repository_test.dart after the
  removal of deprecated prompts.dart file
  - Removed obsolete buildPrompt test group and added new tests for buildTaskDetailsJson method

  5. Documentation Updates

  - Updated /Users/mn/github/lotti/lib/features/ai/README.md to document the automatic title extraction feature
  - Removed references to the deleted prompts.dart file
  - Added detailed explanation of how title extraction works

  Technical Details

  Title Extraction Flow:
  1. AI generates a summary starting with # Suggested Title
  2. Post-processing extracts the title using regex
  3. If current task title < 5 characters, it's replaced with the AI suggestion
  4. The task entity is updated in the database
  5. When displaying the summary, the H1 title is filtered out

  Why This Approach:
  - Stores the complete AI response (including title) for data integrity
  - Filters at display time to avoid UI redundancy
  - Only replaces very short titles to avoid overwriting meaningful user-entered titles

## Testing

  - All existing tests pass
  - Added new tests for title extraction and display filtering
  - Tested with various title formats and edge cases

## Impact

  This feature significantly improves the user experience when creating tasks from audio recordings, as users no
  longer need to manually enter titles - the AI will suggest appropriate titles based on the task content.

### CODE REVIEW

File: test/features/ai/repository/ai_input_repository_test.dart [hunk 1]
```diff
@@ -963,31 +880,30 @@ void main() {
...
-
-        // Check action items in JSON
-        expect(result, contains('"title": "Test Checklist Item"'));
-        expect(result, contains('"completed": true'));
 
-        // Check log entries in JSON
-        expect(result, contains('"text": "Test Journal Entry"'));
-        expect(result, contains('"loggedDuration": "00:30"'));
+        // Parse JSON to verify structure
+        final jsonData = jsonDecode(result!);
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply test/features/ai/repository/ai_input_repository_test.dart]: changed

File: lib/features/ai/util/preconfigured_prompts.dart [hunk 2]
```diff
@@ -58,7 +58,12 @@ Imagine the user has not been involved in the task for a long time, and you want
...
 Also note when the task is done. Note any learnings or insights that can be drawn from the task, if anything is significant. 
 Talk to the user directly, instead of referring to them as "the user" or "they". 
-Don't start with a greeting, don't repeat the task title, get straight to the point. 
+
+Start with a single H1 header (# Title) that suggests a concise, descriptive title 
+for this task. The title should be a single line, ideally under 80-100 characters. 
+If the task already has a title, suggest an improved version that better captures 
+the essence of the task based on the details and logs. Use only one H1 in the 
+entire response. After this, get straight to the point, e.g. no greetings. 
 Keep it short and succinct. Assume the task title is shown directly above in the UI, so starting with the title is not necessary and would feel redundant.
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply lib/features/ai/util/preconfigured_prompts.dart]: adapted

File: lib/features/ai/repository/unified_ai_inference_repository.dart [hunk 3]
```diff
@@ -404,8 +404,30 @@ $response
...
+          final titleMatch = titleRegex.firstMatch(response);
+
+          if (titleMatch != null) {
+            final suggestedTitle = titleMatch.group(1)?.trim();
+            final currentTitle = entity.data.title;
+
+            // Update title if current title is empty or very short (less than 5 characters)
+            if (suggestedTitle != null &&
+                suggestedTitle.isNotEmpty &&
+                currentTitle.length < 5) {
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply lib/features/ai/repository/unified_ai_inference_repository.dart]: extracted

File: lib/features/ai/README.md [hunk 4]
```diff
@@ -1,56 +1,219 @@
...
+)
 ```
 
+## Common Questions
+
+### Q: How are prompts customized?
+A: Prompts use a template system with variables like `{{task}}`. The system replaces these with actual data before sending to the AI.
+
+### Q: Can I extract task titles from summaries?
+A: Currently, the system doesn't extract titles from AI summaries. Summaries are displayed as-is, and titles must be entered manually.
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply lib/features/ai/README.md]: resolved

**Comment by cyberk1ng** [comment]
LGTM
